web/kingpinflag/flag.go

```diff
@@ -33,7 +33,7 @@ func AddFlags(a *kingpin.Application, defaultAddress string) *web.FlagConfig {
 	flags := web.FlagConfig{
 		WebListenAddresses: a.Flag(
 			"web.listen-address",
-			"Addresses on which to expose metrics and web interface. Repeatable for multiple addresses.",
+			"Addresses on which to expose metrics and web interface. Repeatable for multiple addresses. Examples: `:9100` or `127.0.0.1:9100` for http, `vsock://:9100` for vsock",
```

Comment:
I think, vsock having an URI-like format is confusing and asymmetric, especially given that it doesn't really support the host portion of the URI. When a user sees that listening on a vsock expects `vsock://`, they would assume that listening on a TCP port expects `--web.listen-address=http://:9100`, but that's not the case.
Moreover, this help message is confusing, because the vsock connection also uses http. In terms of network layers, vsock is most similar to IP.

I would recommend to change the expected format to `--web.listen-address=vsock:9100`, to avoid confusion.

Reply:
I don't think I like the `vsock:9100` format. I much prefer the URI format.